compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4625,56 +4662,150 @@ private STNode parseObjectConstructorExpression(STNode annots) {
-                    }
-                    break;
-                case ABSTRACT_KEYWORD:
-                    qualifier = consume();
-                    errorCode = DiagnosticErrorCode.ERROR_ABSTRACT_QUALIFIER_NOT_ALLOWED_IN_OBJECT;
+                    nextQualifier = consume();
                     break;
                 case READONLY_KEYWORD:
-                    qualifier = consume();
-                    errorCode = DiagnosticErrorCode.ERROR_READONLY_QUALIFIER_NOT_ALLOWED_IN_OBJECT;
-                    break;
+                case ABSTRACT_KEYWORD:
+                case ISOLATED_KEYWORD:
+                    // Here we allow parsing of old object type qualifiers (`abstract` and `readonly`)
+                    // and then log an error
+                    nextQualifier = consume();
+                    updateLastNodeInListWithInvalidNode(qualifiers, nextQualifier,
+                            DiagnosticErrorCode.ERROR_QUALIFIER_NOT_ALLOWED, nextQualifier.text());
+                    continue;
                 case OBJECT_KEYWORD:
                 default:
                     return STNodeFactory.createNodeList(qualifiers);
             }
+            qualifiers.add(nextQualifier);
+        }
 
-            if (errorCode != null) {
-                if (qualifiers.size() > 0) {
-                    updateLastNodeInListWithInvalidNode(qualifiers, qualifier, errorCode);
-                } else {
-                    addInvalidNodeToNextToken(qualifier, errorCode);
-                }
-                errorCode = null;
+        return STNodeFactory.createNodeList(qualifiers);
+    }
+
+    /**
+     * <p>
+     * Parse object type descriptor qualifiers.
+     * </p>
+     * <code>object-type-descriptor-qualifiers := (client | isolated)*</code>
+     *
+     * @return Parsed node
+     */
+    private STNode parseObjectTypeQualifiers() {
+        STToken nextToken = peek();
+        STToken firstQualifier;
+        switch (nextToken.kind) {
+            case CLIENT_KEYWORD:
+            case ISOLATED_KEYWORD:
+                firstQualifier = consume();
+                break;
+            case READONLY_KEYWORD:
+            case ABSTRACT_KEYWORD:
+                // Here we allow parsing of old object type qualifiers (`abstract` and `readonly`)
+                // and then log an error
+                firstQualifier = consume();
+                addInvalidNodeToNextToken(firstQualifier, DiagnosticErrorCode.ERROR_QUALIFIER_NOT_ALLOWED,
+                        firstQualifier.text());
+                return parseObjectTypeQualifiers();
+            case OBJECT_KEYWORD:
+                return STNodeFactory.createEmptyNodeList();
+            default:
+                recover(nextToken, ParserRuleContext.OBJECT_TYPE_QUALIFIER);
+                return parseObjectTypeQualifiers();
+        }
+
+        return parseObjectTypeNextQualifiers(firstQualifier);
+    }
+
+    private STNode parseObjectTypeNextQualifiers(STNode firstQualifier) {
```

Comment:
Similar to the prev one, I think we don't need the `parseObjectTypeQualifiers()` method